bob/src/cluster/support_types.rs

```diff
@@ -128,3 +128,27 @@ impl IndexMap {
         });
     }
 }
+
+pub(crate) struct NodeOutputJoinHandle<Err>(JoinHandle<Result<NodeOutput<()>, NodeOutput<Err>>>, NodeName);
+
+impl<Err> NodeOutputJoinHandle<Err> {
+    pub(super) fn new(handle: JoinHandle<Result<NodeOutput<()>, NodeOutput<Err>>>, node_name: NodeName) -> Self {
+        Self(handle, node_name)
+    }
+}
+
+impl<Err> Future for NodeOutputJoinHandle<Err> {
+    type Output = Result<Result<NodeOutput<()>, NodeOutput<Err>>, JoinError>;
```

Comment:
But the idea was to propagate the node name in case of `JoinError`. Right now this future basically do nothing: it replaces node name, but the original node name inside is already the correct one.

The proper output type should be `Result<NodeOutput<()>, NodeOutput<Err>>`